src/include/migraphx/op/nonmaxsuppression.hpp

```diff
@@ -33,7 +33,6 @@ struct nonmaxsuppression
     shape compute_shape(std::vector<shape> inputs) const
     {
         // requires at least 2 inputs
-        check_shapes{inputs, *this}.standard();
```

Comment:
What kind of non-standard shapes do you expect to see ? 

If they are non-standard i.e. let's say transpose then, boxes could be = [num_of_boxes, batches, 4]. We would need to update reference indices to these dimensions everywhere in the code.

Reply:
Yeah, this code is not going to handle non-standard shapes correctly. I think the reason why it works for the resnet34-ssd model is because our `auto_contiguous{}` pass makes the shape standard before it gets to this instruction. However, with the standard shape check the model doesn't get to that pass. I'll look into what I can do to use the strides rather than the fixed offsets.

Reply:
> > If they are non-standard i.e. let's say transpose then, boxes could be = [num_of_boxes, batches, 4]
> 
> The dimensions should be the same order regardless of the order in memory. If they store the dimensions as `[num_of_boxes, batches, 4]` in memory, then a transpose will be needed to put the dimensions in the correct order of `[num_batches, spatial_dimension, 4]`. Thus we would need to support a transposed shape.

Agree, i am just saying current changes are not accounting for the fact that inputs could be transposed. 

e.g.  hardcoding following
```
        auto batch_num   = lens[0];
        auto class_num   = lens[1];
        auto box_num     = args.at(0).get_shape().lens()[1];
```

Reply:
> > If they are non-standard i.e. let's say transpose then, boxes could be = [num_of_boxes, batches, 4]
> 
> The dimensions should be the same order regardless of the order in memory. If they store the dimensions as `[num_of_boxes, batches, 4]` in memory, then a transpose will be needed to put the dimensions in the correct order of `[num_batches, spatial_dimension, 4]`. Thus we would need to support a transposed shape.

Would that mean, MIGraphX would have to insert extra transposes to make shape requirement adhere to [num_batches, spatial_dimension, 4]` ? Do we do that currently ? I guess no.

Reply:
> Agree, i am just saying current changes are not accounting for the fact that inputs could be transposed.

Yea, but when we say the shape is transposed we mean the memory layout is transposed not the dimensions.

> Would that mean, MIGraphX would have to insert extra transposes to make shape requirement adhere to [num_batches, spatial_dimension, 4] ?

Well the user would insert the transposes as necessary to get the dimensions in the correct order. This could happen right before the call but it could be precede several other shape manipulations as well.